CONTRIBUTING.md

```diff
@@ -1,37 +1,74 @@
-# Contributing to status-go
+# Contributing Guidelines
 
-First of all, thank you for taking time to add more value to Status, we really appreciate it!
+Thank you for considering contributing to our project! We appreciate your time and effort to make this project better.
 
-If you just have a question, don't open an issue but rather ask us on our [Discord server](https://discord.gg/3Exux7Y)
-## Starter Links
-
-You just want to contribute something without reading tons of documentation, right? There're only a few useful links to start with.
-
-How [status-mobile](github.com/status-im/status-mobile) uses us:
-https://github.com/status-im/status-go/wiki/Notes-on-Bindings
-
-Architecture: TBD in [#238](https://github.com/status-im/status-go/issues/238)
-
-You can also discover more information in https://hackmd.io/s/SkZI0bXIb
+## Table of Contents
+1. [Workflow](#workflow)
+2. [Code Style](#code-style)
+3. [Keep History Clean](#keep-history-clean)
+4. [Testing](#testing)
+   - [Test Validation](#test-validation)
+   - [Area of Impact](#area-of-impact)
+5. [Pull Request Description](#pull-request-description)
+   - [Feature Flags](#feature-flags)
+   - [Removing Feature Flags](#removing-feature-flags)
+6. [Test Coverage](#test-coverage)
+   - [Maintaining Test Coverage](#maintaining-test-coverage)
+   - [Metrics for Test Coverage](#metrics-for-test-coverage)
 
 ## Workflow
 
 1. Pick an [issue](https://github.com/status-im/status-go/issues) to work on and drop a line there that you're working on that.
-2. Wait for an approve from one of core contributors.
+2. Wait for approval from one of the core contributors.
 3. Fork the project and work right in the `develop` branch.
 4. Work on the issue and file a PR back into `develop`.
 5. Wait until your PR is [reviewed](https://hackmd.io/s/B1AenvFU-) by 2 core developers and merged.
 
 ## Code Style
 
-Please, note that we follow [Effective Go](https://golang.org/doc/effective_go.html) and
-[CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) in our code.
+Please note that we follow [Effective Go](https://golang.org/doc/effective_go.html) and [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments) in our code.
+
+## Keep History Clean
+
+1. **Squash PR before merging**: You can do it either with GitHub API by merging with "Squash and merge" or locally if you want to preserve your signature. It is ok to merge multiple commits with "Rebase and merge" if they are logically separate.
+
+2. **Preserve as much history as possible**: If you need to re-name a file, use `git mv` - it will preserve git history.
+
+## Testing
+
+### Test Validation
+
+Every Pull Request (PR) should include tests to validate its correctness and to test the features implemented. Preferably, use Behavior-Driven Development (BDD) principles for writing tests.
```

Comment:
while BDD is fairly common perhaps there's a useful link for this?

Reply:
updated